libfive/include/libfive/oracle/oracle_clause.hpp

```diff
@@ -54,8 +54,20 @@ class OracleClause
     {
         return {};
     }
+
     unsigned rank() const;
 
+    virtual std::unique_ptr<OracleClause> clone() const = 0;
```

Comment:
not sure I like this , because it requires all our oracles to implement this , and some implementations might do deep copies making things inefficient

Reply:
I went through all oracles and they all basically wrap a shared pointer around the actual oracle (see the last commit in the ntimp pr). But I agree it’s a bit inconvenient.